src/main/java/com/assignment1/Runner.java

```diff
@@ -0,0 +1,117 @@
+package com.assignment1;
+
+import com.assignment1.exceptions.RuntimeExceptionCustom;
+import com.assignment1.item.ItemEntity;
+import com.assignment1.item.TaxCalculator;
+import com.assignment1.item.Taxable;
+import com.assignment1.parser.Valid;
+import com.assignment1.parser.Validator;
+import com.assignment1.userinterface.UserInterface;
+import com.assignment1.userinterface.UserInterfaceHandler;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.InputMismatchException;
+import java.util.List;
+
+/**
+ * Runner class having main function to run the application.
+ * (Acting as controller)
+ **/
+@SuppressWarnings({"PMD.SystemPrintln"})
+public final class Runner {
+
+  /** ArrayList to store the Items in order to display later. */
+  private static final List<ItemEntity> ITEM_ARRAY = new ArrayList<>();
+
+  /** YES Constant for input as 'y' */
+  private static final String RESPONSE_YES = "y";
+  /** NO Constant for input as 'n' */
+  private static final String RESPONSE_NO = "n";
+
+  //Private Constructor so that it can't be instantiated.
+  private Runner() {
+  }
+
+  //Method to process the input given on starting application
+  private static void processCommandLineArgs(final Valid validObj,
+                              final UserInterfaceHandler uiObj,
+                              final Taxable taxObj,
+                              final String... args) {
+    ItemEntity item;
+    double taxOutput;
+    if (args.length > 0) {
+      try {
+        item = validObj.validate(args); //Parse the initial Command Line Arguments
```

Comment:
why is validate doing both `validation` and `parsing`? it should do only one thing

Reply:
think about this, some methods can be simplified if you do this

Reply:
Incorporated the change. Used an Intermediate object (ItemDTO) and made validate(ItemDTO) return boolean after only validation.